runtime/src/sudo_wrapper.rs

```diff
@@ -0,0 +1,98 @@
+use codec::{Decode, DecodeWithMemTracking, Encode};
+use frame_support::dispatch::{DispatchInfo, PostDispatchInfo};
+use frame_support::pallet_prelude::Weight;
+use frame_support::traits::IsSubType;
+use frame_system::Config;
+use pallet_sudo::Call as SudoCall;
+use scale_info::TypeInfo;
+use sp_runtime::traits::{
+    AsSystemOriginSigner, DispatchInfoOf, Dispatchable, Implication, TransactionExtension,
+    ValidateResult,
+};
+use sp_runtime::transaction_validity::{
+    InvalidTransaction, TransactionSource, TransactionValidityError,
+};
+use sp_std::marker::PhantomData;
+use subtensor_macros::freeze_struct;
+
+#[freeze_struct("99dce71278b36b44")]
+#[derive(Default, Encode, Decode, DecodeWithMemTracking, Clone, Eq, PartialEq, TypeInfo)]
+pub struct SudoTransactionExtension<T: Config + Send + Sync + TypeInfo>(pub PhantomData<T>);
+
+impl<T: Config + Send + Sync + TypeInfo> sp_std::fmt::Debug for SudoTransactionExtension<T> {
+    #[cfg(feature = "std")]
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(f, "SudoTransactionExtension",)
+    }
+    #[cfg(not(feature = "std"))]
+    fn fmt(&self, _: &mut core::fmt::Formatter) -> core::fmt::Result {
+        Ok(())
+    }
+}
+
+impl<T: Config + Send + Sync + TypeInfo> SudoTransactionExtension<T> {
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+}
+
+impl<T: Config + Send + Sync + TypeInfo + pallet_sudo::Config>
+    TransactionExtension<<T as Config>::RuntimeCall> for SudoTransactionExtension<T>
+where
+    <T as Config>::RuntimeCall: Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+    <T as Config>::RuntimeOrigin: AsSystemOriginSigner<T::AccountId> + Clone,
+    <T as Config>::RuntimeCall: IsSubType<SudoCall<T>>,
+{
+    const IDENTIFIER: &'static str = "SudoTransactionExtension";
+
+    type Implicit = ();
+    type Val = Option<T::AccountId>;
```

Comment:
```suggestion
    type Val = ();
```

I get the impression the `Val` is never used? So may be better to just use unit type.

It is only used if you have a `prepare` implementation because it is passed as a parameter.

Reply:
I also use it in return type for validate.

Reply:
> I also use it in return type for validate.

Yes, but the [`validate` return value is only used in the `prepare`](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_frame/traits/trait.TransactionExtension.html#associatedtype.Val) function and because there is no implementation here, it is not used at all so you can just remove it from `Val` and from return type of `validate`